src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -45,12 +46,30 @@ public void storeMessage(Message message) {
     datastore.put(messageEntity);
   }
 
-    /** Stores FoodItem in Datastore. */
+    /** Stores FoodItem in Datastore.
+      * Prevent the same food from being entered into the database more than once
+      */
+
   public void storeFood(FoodItem food) {
-    Entity foodEntity = new Entity("FoodItem", food.getID().toString());
-    foodEntity.setProperty("Name", food.getName());
-    foodEntity.setProperty("CO2", food.getCO2PerYear());
-    datastore.put(foodEntity);
+    if (exists(food) == true){
+       return;
+    }
+    else {
+      Entity foodEntity = new Entity("FoodItem", food.getID().toString());
+      foodEntity.setProperty("Name", food.getName());
+      foodEntity.setProperty("CO2", food.getCO2PerYear());
+      datastore.put(foodEntity);
+      return;
+    }
+  }
+
+  public boolean exists(FoodItem onefood) {
```

Comment:
`onefood` is a bit of an odd name, but that's just my opinion. It should be `oneFood` however.